CMakeLists.txt

```
@@ -84,11 +91,11 @@ set(SOURCE_FILES
 source_group(include FILES ${INCLUDE_FILES})
 source_group(source FILES ${SOURCE_FILES})
 
-add_library(enet STATIC
+add_library(enet ${ENET_BUILD_LIBTYPE}
```

```suggestion
add_library(enet
```

Not necessary, CMake will automatically use the value of `BUILD_SHARED_LIBS`

https://cmake.org/cmake/help/latest/command/add_library.html